src/datamap.rs

```diff
@@ -62,5 +72,21 @@ impl DataMap {
     }
     return size
   }
+  // using bubble sort
+  // TODO : change to use other quick sort algorithm to improve the performance
+  fn chunks_sort(chunks: &mut [ChunkDetails]) {
+    let (mut i, len) = (0, chunks.len());
+    while i < len {
+        let (mut j, mut cur_min) = (i + 1, i);
+        while j < len {
+            if chunks[j].pre_hash < chunks[cur_min].pre_hash {
+                cur_min = j;
+            }
+            j = j + 1;
+        }
+        chunks.swap(i, cur_min);
+        i = i + 1;
+    }
   }
+}
```

Comment:
I think the previous implementation of sort trait is incorrect and won't work as expected.
Actually it is get_sorted_chunks needs to be implemented ( and got implemented in this pull request)